src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ServiceJsonRpcDescriptorWithNuGetCoreConverters.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using StreamJsonRpc;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    internal class ServiceJsonRpcDescriptorWithNuGetCoreConverters : ServiceJsonRpcDescriptor
+    {
+        internal ServiceJsonRpcDescriptorWithNuGetCoreConverters(ServiceMoniker serviceMoniker, MessageDelimiters messageDelimiter)
+            : base(serviceMoniker, Formatters.MessagePack, messageDelimiter)
+        {
+        }
+
+        internal ServiceJsonRpcDescriptorWithNuGetCoreConverters(ServiceMoniker serviceMoniker, Type clientInterface, MessageDelimiters messageDelimiter)
+            : base(serviceMoniker, clientInterface, Formatters.MessagePack, messageDelimiter)
+        {
+        }
+
+        protected ServiceJsonRpcDescriptorWithNuGetCoreConverters(ServiceJsonRpcDescriptorWithNuGetCoreConverters copyFrom)
+            : base(copyFrom)
+        {
+        }
+
+        protected override ServiceRpcDescriptor Clone() => new ServiceJsonRpcDescriptorWithNuGetCoreConverters(this);
+
+        protected override IJsonRpcMessageFormatter CreateFormatter()
+        {
+            Assumes.True(Formatter == Formatters.MessagePack);
```

Comment:
This contradicts everything else. You're using Newtonsoft everywhere: for the rest of this method and your converter itself. So if your formatter is MessagePack, the rest of it needs to be updated to use MessagePack formatters.